// @forge/manifest/out/validators/remotes-is-configurable-validator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemotesIsUserConfigurableValidator = void 0;
const text_1 = require("../text");
const utils_1 = require("../utils");
class RemotesIsUserConfigurableValidator {
    async validateRemote(manifest) {
        if (!manifest?.typedContent?.remotes) {
            return {
                success: true,
                manifestObject: manifest
            };
        }
        const remotes = manifest.typedContent.remotes;
        const validationErrors = [];
        remotes?.forEach((remote) => {
            const hasConfigurable = remote?.configurable !== undefined;
            if (!hasConfigurable && remote.baseUrl === undefined) {
                validationErrors.push({
                    message: text_1.errors.modules.remote.neitherBaseUrlNorUserConfigurationPresent(remote.key),
                    reference: text_1.References.App,
                    level: 'error',
                    ...(0, utils_1.findPosition)(remote.key, manifest.yamlContentByLine)
                });
            }
        });
        return {
            success: validationErrors.length === 0,
            manifestObject: manifest,
            errors: validationErrors
        };
    }
    async validate(manifest) {
        return this.validateRemote(manifest);
    }
}
exports.RemotesIsUserConfigurableValidator = RemotesIsUserConfigurableValidator;
